webapp/components/_new/generic/ResponsiveImage/ResponsiveImage.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <img :data-src="lazySrc" :data-srcset="lazySrcset" :style="style" class="responsive-image lozad" />
+</template>
+
+<script>
+import lozad from 'lozad'
+
+export default {
+  name: 'ResponsiveImage',
+  props: {
+    backgroundColor: {
+      type: String,
+      default: '#efefef',
+    },
+    height: {
+      type: Number,
+      default: null,
+    },
+    lazySrc: {
+      type: String,
+      default: null,
+    },
+    lazySrcset: {
+      type: String,
+      default: null,
+    },
+    width: {
+      type: Number,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+    }
+  },
+  computed: {
+    aspectRatio() {
+      if (!this.width || !this.height) return null
+      return (this.height / this.width) * 100
+    },
+    style() {
+      const style = { backgroundColor: this.backgroundColor }
+      if (this.width) style.width = `${this.width}px`
+      const applyAspectRatio = this.loading && this.aspectRatio
+      if (applyAspectRatio) {
+        style.height = 0
+        style.paddingTop = `${this.aspectRatio}%`
+      }
+
+      return style
+    },
+  },
+  mounted() {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-28T22:59:44Z" title="Tuesday, January 28th 2020, 11:59:44 pm +01:00">Jan 28, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,87 @@
+<template>
+  <img :data-src="lazySrc" :data-srcset="lazySrcset" :style="style" class="responsive-image lozad" />
+</template>
+
+<script>
+import lozad from 'lozad'
+
+export default {
+  name: 'ResponsiveImage',
+  props: {
+    backgroundColor: {
+      type: String,
+      default: '#efefef',
+    },
+    height: {
+      type: Number,
+      default: null,
+    },
+    lazySrc: {
+      type: String,
+      default: null,
+    },
+    lazySrcset: {
+      type: String,
+      default: null,
+    },
+    width: {
+      type: Number,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+    }
+  },
+  computed: {
+    aspectRatio() {
+      if (!this.width || !this.height) return null
+      return (this.height / this.width) * 100
+    },
+    style() {
+      const style = { backgroundColor: this.backgroundColor }
+      if (this.width) style.width = `${this.width}px`
+      const applyAspectRatio = this.loading && this.aspectRatio
+      if (applyAspectRatio) {
+        style.height = 0
+        style.paddingTop = `${this.aspectRatio}%`
+      }
+
+      return style
+    },
+  },
+  mounted() {
```


I would love to avoid the event listeners here. State == root of all evil.